docs/migration-console.md

```diff
@@ -0,0 +1,123 @@
+
+The overall architecture can be seen in this diagram. Each of the subsections below discusses a component in more detail, moving from left (user) to right (deployed services). Green indicates examples, not definite implementations.
+
+![Library Diagram](diagrams/migration-console-library.svg)
+
+[source](diagrams/migration-console-library-source.md) (This is an [excalidraw](https://github.com/excalidraw/excalidraw) markdown file, not particularly human readable.)
+
+### Interfaces
+
+The frontends currently envisioned are a command line interface (CLI) and a web API. While these are not explicitly part of the library, they are closely linked and discussing them is helpful context for the design decisions of the library interfaces.
+
+#### CLI
+
+In the near term of the Migration Assistant, the CLI is the primary way for users to manage and understand their migration. After deploying their tools, a user will log onto the Migration Console and use the CLI for tasks like starting and checking the progress of their historic backfill, turning on and off the replayer, running `_cat/indices` against the source and target cluster, etc. These functions will be performed in the same way by the user, regardless of whether their tools are deployed locally on Docker, remotely on AWS, a different hosting platform, or a combination thereof.
+
+#### Web API
+
+Parallel to the CLI, there will be a web API to access the same functionality. A web API to access the full functionality, with authorization, etc., is not on our roadmap yet, but this is envisioned as being a powerful way for users to access and manage their migration from their own environment, potentially programatically. We could also use the web API to build a web frontend with UI, etc. down the line.
+
+##### DMS API
+
+In the short term, a small subset of the functionality is necessary to meet the requirements of our in-progress collaboration with Database Migration Service (DMS). In some of these cases, DMS has specified an API contract that doesn't align with our plan for the API endpoints and data received.  In these cases, we will build out DMS specific endpoints that meet these exact purposes without trying to also match the general case.
+
+#### Generation from a spec
+
+For the CLI and (standard) web API, the intention is that the “frontend” should be very small—essentially just the necessary code to get the user’s data from the input source (either CLI commands or http requests) into the library itself. To accomplish this, the actual endpoints in both cases could be generated from an API spec or one of them from the other. This will be a helpful way to minimize the developer effort to support additional commands and ensure that behavior is consistent between the two modalities.
+
+#### Custom Frontends
```

Comment:
What do you think about removing this section until we support a scenario like this?  If we don't already have this captured in a jira I think this would be a good candidate for one so the idea isn't lost (I really like the idea!)

Reply:
We actually do have one--the integ tests. I changed this to do a clearer job describing that and de-emphasized the future ideas. Let me know if you want them fully removed.